content/blog/deploy-stateful-mongodb-application-on-kubernetes-clusters-in-hpe-greenlake-for-containers-1.md

```diff
@@ -0,0 +1,547 @@
+---
+title: Deploy Stateful MongoDB Application on Kubernetes Clusters in HPE
+  GreenLake for Containers
+date: 2022-08-16T14:02:17.749Z
+author: Guoping Jia
+authorimage: /img/guoping.png
+tags:
+  - hpe-greenlake
+  - kubernetes
+  - mongodb
+---
+## Introduction
+
+[HPE GreenLake for Containers](https://www.hpe.com/us/en/greenlake/containers.html), one of the HPE GreenLake cloud services available on the HPE GreenLake Central platform, allows customers to open the Clusters screen to create a cluster, view details about existing clusters, and launch the HPE GreenLake for Container service console. It provides an enterprise-grade container management service using open source Kubernetes. 
+
+This blog post describes the process of deploying the stateful MongoDB application on the created Kubernetes clusters in HPE GreenLake for Containers. Using the Kubernetes StatefulSet and Headless Service, together with the pre-provisioned persistent volumes, the MongoDB application can be deployed as a Replica Set that provides redundancy, fault tolerance and high availability. This MongoDB application deployment can be used for working with a large amount of data and high loads in customer production environments.
```

Comment:
I recommend changing "the stateful MongoDB" to "a stateful MongoDB"
Shouldn't "Replica Set" be "ReplicaSet"? (that's how I saw it stated in a number of places on Google :-) )
I'm not sure what you mean by "high loads in customer production environments." Should it be "and a high number of workloads in customer production environments."?

Reply:
"ReplicaSet" is used most often in Kubernetes as an API object (at https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/), while "Replica Set" is a definition used especially in MongoDB referring to its replication architecture (at https://www.mongodb.com/docs/manual/reference/glossary/#std-term-replica-set). It does cause some confusion when I first tried it on Kubernetes. So I choose to use the "Replica Set" to comply with MongoDB manual to avoid any confusion with Kubernetes.

Reply:
I'm good with that. Thanks for the clarification!